use-cases/inferencing/serving-with-vllm/README.md

```diff
@@ -0,0 +1,513 @@
+  sed -i -e "s|_VOLUME-HANDLE_|${VOLUME-HANDLE}|" manifests/volume-prep/pv-and-pvc.yaml
+  kubectl apply -f manifests/volume-prep/pv-and-pvc.yaml
+  ```
+
+## Deploy a vLLM container to your cluster
+
+- Run the batch job to deploy model using persistent disk on GKE.
+  There are few variables that need to be updated before you can run this job to download model on your persistent disk
+
+
+  Here is an example 
+    ```
+    mkdir -p /mnt/model-gemma2-a100
+    cp -r /data/models/model-gemma2-a100/experiment-a2aa2c3it1 /mnt/model-gemma2-a100  # location to copy the model artifacts from
+    ls -lR /mnt/model-gemma2-a100 # location to copy the model artifacts
+    ````
+
+  ```sh
+  MODEL-ID= "<model-id>"
+  MODEL-DIR-PATH ="<LOCAl-PATH-TO-MODEL_DIR>" # location to copy the model artifacts from
+  sed -i -e "s|_MODEL-ID_|${MODEL-ID}|" manifests/volume-prep/batch-job-model-deployment.yaml
+  sed -i -e "s|_MODEL-DIR-PATH_|${MODEL-DIR-PATH}|" manifests/volume-prep/batch-job-model-deployment.yaml
+  sed -i -e "s|_NAMESPACE_|${NAMESPACE}|" manifests/volume-prep/batch-job-model-deployment.yaml
+  sed -i -e "s|_ACCELERATOR_TYPE_|${ACCELERATOR_TYPE}|" manifests/volume-prep/batch-job-model-deployment.yaml
+  ```
+
+  ```sh
+  MODEL-ID= <model-id>
+  sed -i -e "s|_MODEL-ID_|${MODEL-ID}|" manifests/model_deployment.yaml
+  kubectl create -f manifests/model_deployment.yaml
+  kubectl logs -f -l app=vllm-openai -n ${NAMESPACE}
+  ```
+
+  ```sh
+  INFO:     Started server process [1]
+  INFO:     Waiting for application startup.
+  INFO:     Application startup complete.
+  INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+  ```
+
+## Serve the deployed model through curl and a web chat interface
+
+- Test your deployed model through the CLI
+
+  ```sh
+  kubectl port-forward svc/vllm-openai -n ${NAMESPACE} 8000
+  ```
+
+  Run the curl prompt with your values
+
+  ```sh
+  USER_PROMPT="I'm looking for comfortable cycling shorts for women, what are some good options?"
+  MODEL_ID=""
+
+  curl http://localhost:8000/v1/chat/completions \
+    -H "Content-Type: application/json" \
+    -d '{
+        "model": "${MODEL_ID}",
+        "messages": [
+            {"role": "user", "content": "${USER_PROMPT}"}],
+             "temperature": 0.70,
+             "top_p": 1.0,
+             "top_k": 1.0,
+             "max_tokens": 256
+    }'  
+  ```
+
+- You can also deploy a gradio chat interface to view the model chat interface. [OPTIONAL]
+  
+  ```sh
+  MODEL-ID=<your-model-id>
+  sed -i -e "s|_MODEL-ID_|${MODEL-ID}|" manifests/gradio.yaml
+  ```
+
+  ```sh
+  sed -i -e "s|_NAMESPACE_|${NAMESPACE}|" manifests/gradio.yaml
+  kubectl apply -f manifests/gradio.yaml
+  ```
+
+### Production Metrics
+
+vLLM exposes a number of metrics that can be used to monitor the health of the system. These metrics are exposed via the `/metrics` endpoint on the vLLM OpenAI compatible API server.
+
+  ```sh
+  curl http://localhost:8000/metrics
```

Comment:
is this in reference to an example, since we can't kubectl exec on the platform

Reply:
Do we know if these metrics are scrapable by GMP?

Reply:
Yes, they are captured if GMP and pod monitoring is enabled